html/user/openssl_sign_action.php

```diff
@@ -0,0 +1,134 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+
+// Needs to be defined, unknown use.
+check_get_args(array("tnow", "ttok"));
+
+// Check the user is online
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+// For some reason, I'm only able to use numbers, no letters.
+// Need to provide security, but full functionality is likewise desirable.
+$user_data = sanitize_tags(post_str("user_data", true));
+
+// Following line a potential bug.
+// $user_data = BoincDb::escape_string($user_data);
+
+//Has the user entered data with length > 0 & <= 256?
+if (($user_data) && (strlen($user_data) <= 256)) {
+	// Retrieve the logged in user's UserID.
+	$userid = $user->id;
+
+	// Create the message which will be signed.
+	$message_data = "$userid $user_data";
```

Comment:
Missing the timestamp from your proposal @grctest.

Reply:
I changed my proposal after your reply on steemit, the timestamp was removed because:
* The external system key should be sufficient to prevent cross-network beacon replay attacks (stealing a registration from A to register on B).
* Since UserID doesn't change, the beacon containing this 2-way handshake data should be permanent. If a beacon is invalidated due to external network issues, then perhaps the beacon could be replayed but if the old beacon isn't available for comparison then a timestamp wouldn't stop a replay. Ultimately they could replay but they couldn't change the external system key, so an attacker would just be re-advertising your legit beacon for you (not really an attack).

Think the timestamp should be reintroduced?